eng/common/scripts/Helpers/git-helpers.ps1

```diff
@@ -0,0 +1,39 @@
+# cSpell:ignore Committish
+# cSpell:ignore committish
+# cSpell:ignore PULLREQUEST
+# cSpell:ignore TARGETBRANCH
+# cSpell:ignore SOURCEVERSION
+function Get-ChangedFiles {
+  param (
+    [string]$SourceCommittish= "${env:BUILD_SOURCEVERSION}",
```

Comment:
@sima-zhu have you done any testing with using BUILD_SOURCEVERSION vs SYSTEM_PULLREQUEST_SOURCECOMMITID? When I was checking your test runs I noticed those sha's are not the same and I was expecting to see the value in SYSTEM_PULLREQUEST_SOURCECOMMITID. I believe BUILD_SOURCEVERSION is the merge commit which is not what we want in most of the cases. However I know there was some investigation when trying out `..` vs `...` so wasn't sure if you investigated that further or not.

@benbp I'd be interested in your thoughts as well.

Reply:
I will add I know we hit issues with the eng-common-workflow a few times when we were comparing main to HEAD (I believe HEAD is generally set to BUILD_SOURCEVERSION) whenever there were other eng/common changes merged into main between the time the merge commit was created and when the diff was produced.

Reply:
The reason I used BUILD_SOURCEVERSION is this is what pipeline locally have in current working branch. Both SYSTEM_PULLREQUEST_SOURCECOMMITID and BUILD_SOURCEVERSION return the same based on my testing. I don't have a strong preference since we only used in PR, so SYSTEM_PULLREQUEST_SOURCECOMMITID always exists

Reply:
I agree to use SYSTEM_PULLREQUEST_SOURCECOMMITID since the value is there from PR. source version is something generated later.

Reply:
Yes I think SYSTEM_PULLREQUEST_SOURCECOMMITID is the correct thing to use as BUILD_SOURCEVERSION is generally there merge commit which with the `...` operator usually returns the same thing based on some of my testing but I think to be more precise we should use SYSTEM_PULLREQUEST_SOURCECOMMITID.

Reply:
Done.